scripts/ai/AIDriveStrategyFieldWorkCourse.lua

```diff
@@ -582,12 +596,17 @@ function AIDriveStrategyFieldWorkCourse:getTurnEndForwardOffset()
     return 0
 end
 
-function AIDriveStrategyFieldWorkCourse:rememberWaypointToContinueFieldwork()
-    self.storage.continueFieldworkAtWaypoint = self:getBestWaypointToContinueFieldwork()
+function AIDriveStrategyFieldWorkCourse:rememberWaypointToContinueFieldWork()
+    local ix = self:getBestWaypointToContinueFieldwork()
+    self.vehicle:rememberCpLastWaypointIx(ix)
+end
+
+function AIDriveStrategyFieldWorkCourse:getRememberedWaypointToContinueFieldWork()
+    return self.vehicle:getCpLastRememberedWaypointIx()
 end
 
 function AIDriveStrategyFieldWorkCourse:getBestWaypointToContinueFieldwork()
-    local bestKnownCurrentWpIx = self.ppc:getLastPassedWaypointIx() or self.ppc:getCurrentWaypointIx()
+    local bestKnownCurrentWpIx = self.course:getLastPassedWaypointIx() or self.course:getCurrentWaypointIx()

```

Comment:
Not now, but we'll need to keep the fieldwork course separate and track the progress on that, as self.course is dynamic and may contain any temporary course as well.

Reply:
Oh, just saw your commit message, so course is now always the fieldwork course and we set the temporary course on the ppc only.

Reply:
@pvaiko Simply guessed that it might work. Do you think this is good solution ?

Reply:
As long as we agree that self.course is never changed after the work is started, sure. But then we have to make sure that all functions using the current course (like the onWaypointChange) use the actual course PPC is driving and not self.course.